hc/accounts/tests/test_login.py

```diff
@@ -1,37 +1,48 @@
 from django.contrib.auth.models import User
 from django.core import mail
-from django.test import TestCase
+from hc.test import BaseTestCase
 from hc.api.models import Check
 
 
-class LoginTestCase(TestCase):
+class LoginTestCase(BaseTestCase):
 
     def test_it_sends_link(self):
-        check = Check()
+        initial_count = len(User.objects.all())
+        check = Check(user=self.alice)
         check.save()
 
+
         session = self.client.session
         session["welcome_code"] = str(check.code)
         session.save()
+        form = {"email": "adrian@example.org"}
+        response = self.client.post("/accounts/login/", form)
+        assert response.status_code == 302
 
-        form = {"email": "alice@example.org"}
-
-        r = self.client.post("/accounts/login/", form)
-        assert r.status_code == 302
-
-        ### Assert that a user was created
+        # Assert that a user was created
+        ##use count
+        ##assert user exists
+        user = User.objects.get(email=form["email"])
+        final_count = len(User.objects.all())
+        self.assertEqual(final_count, initial_count+1)
 
         # And email sent
         self.assertEqual(len(mail.outbox), 1)
         self.assertEqual(mail.outbox[0].subject, 'Log in to healthchecks.io')
-        ### Assert contents of the email body
 
-        ### Assert that check is associated with the new user
+        # Assert contents of the email body
+        self.assertIn(self.profile.token, mail.outbox[0].body)
+
+        # ## Assert that check is associated with the new user
+        check = Check(user=self.alice)
+        check.save()
+        test_user = User.objects.get(email="alice@example.org")
+        check = Check.objects.get(id=test_user.id)
+        self.assertEqual(check.id, test_user.id)
```

Comment:
This assertion is failing and causing the [build](https://travis-ci.org/andela/healthchecks-thunknalla/jobs/235994362) to fail because the variable `check` on line 37 is being overwritten on line 40. The latter is a queryset containing multiple Check objects created by Alice while the former is the check object that you ought to be asserting that it is associated with Alice. Delete line 40 update the assertion to be
```python
self.assertEqual(check.user.id, test_user.id)
```
Makes sense?